pkg/backporter/backporter.go

```diff
@@ -159,214 +168,264 @@ const clonesTemplateConstructor = `
 				</tr>
 			{{ end }}
 		{{ else }}
-			<tr> <td colspan=4 style="text-align:center;"> No clones found! </td></tr>
+			<tr> <td colspan=4 style="text-align:center;"> No clones found. </td></tr>
 		{{ end }}
 		</tbody>
-	</table>`
+	</table>
+	<form class="form-inline my-2 my-lg-0" role="search" action="/clones/create" method="post">
+		<input type="hidden" name="ID" value="{{.Bug.ID}}">
+		<select class="form-control mr-sm-2" aria-label="Search" name="release" id="target_version" required>
+			<option value="" disabled selected hidden>Target Version</option>
+			{{ range $release := .CloneTargets }}
+				<option value="{{$release}}" id="opt_{{$release}}">{{$release}}</option>
+			{{end}}
+		</select>
+		<button class="btn btn-outline-success my-2 my-sm-0" type="submit">Create Clone</button>
+    </form>
+</div>`
+
+const errorTemplateConstructor = `
+<div class="alert alert-danger" id="error-banner">
+<a href="#" class="close" data-dismiss="alert" aria-label="close">&times;</a>
+<strong>Error </strong> <label id ="error-text">{{.}}</label>
+</div>`
 
 var (
 	clonesTemplate = template.Must(template.New("clones").Parse(clonesTemplateConstructor))
 	emptyTemplate  = template.Must(template.New("empty").Parse("{{.}}"))
+	errorTemplate  = template.Must(template.New("error").Parse(errorTemplateConstructor))
 )
 
+func leanErrorHandler(w http.ResponseWriter, err error, shortErrorMessage string, statusCode int, endpoint string, bugID int) {
+	w.WriteHeader(statusCode)
+	_, fprintfErr := fmt.Fprintf(w, "%s: %v", shortErrorMessage, err)
+	logrus.WithFields(logFieldsFor(endpoint, bugID)).WithError(fmt.Errorf("%s: %v", shortErrorMessage, utilerrors.NewAggregate([]error{err, fprintfErr})))
+}
+
+func errorPageHandler(w http.ResponseWriter, err error, shortErrorMessage string, statusCode int, endpoint string, bugID int) {
```

Comment:
I just ended up reusing the lean handler in  writePage since it did what most of what I wanted there, but I have removed its use from the errorPageHandler and have gone with your second suggestion - ie make writePage return an error.

The lean handler is primarily meant to handle the errors for the getBugHanlder which returns a simple JSON and its error were also intended to be similarly simple while the errorPageHandler returns an entire error page and serves a different purpose I feel. Do you still think one needs to be removed?

I'll rename the functions.

Reply:
Yeah I think one of them should be removed as its not clear which to use when. I don't think it is very important if we have a nice error page or just the error text

Reply:
Ok. Done